File: levelledmobs-plugin-bukkit/src/main/java/io/github/arcaneplugins/levelledmobs/bukkit/command/levelledmobs/subcommand/ReloadSubcommand.java
package io.github.arcaneplugins.levelledmobs.bukkit.command.levelledmobs.subcommand;

import dev.jorel.commandapi.CommandAPICommand;
import io.github.arcaneplugins.levelledmobs.bukkit.LevelledMobs;
import io.github.arcaneplugins.levelledmobs.bukkit.config.translations.Message;

public final class ReloadSubcommand {

    public static CommandAPICommand createInstance() {
        return new CommandAPICommand("reload")
            .withPermission("levelledmobs.command.levelledmobs.reload")
            .withShortDescription("Reloads LM config files.")
            .withFullDescription("Reloads LevelledMobs config files.")
            .executes((sender, args) -> {
                Message.COMMAND_LEVELLEDMOBS_SUBCOMMAND_RELOAD_START.sendTo(sender);

                try {
                    LevelledMobs.getInstance().reload();
                } catch(final Exception ex) {
                    ex.printStackTrace();
                    Message.COMMAND_LEVELLEDMOBS_SUBCOMMAND_RELOAD_COMPLETE_FAILURE.sendTo(sender);
                    return;
                }

                Message.COMMAND_LEVELLEDMOBS_SUBCOMMAND_RELOAD_COMPLETE_SUCCESS.sendTo(sender);
            });
    }

}

File: levelledmobs-plugin/src/main/kotlin/io/github/arcaneplugins/levelledmobs/commands/subcommands/ReloadSubcommand.kt
package io.github.arcaneplugins.levelledmobs.commands.subcommands

import com.mojang.brigadier.Command
import com.mojang.brigadier.tree.LiteralCommandNode
import io.github.arcaneplugins.levelledmobs.LevelledMobs
import io.github.arcaneplugins.levelledmobs.misc.FileLoader
import io.papermc.paper.command.brigadier.CommandSourceStack
import io.papermc.paper.command.brigadier.Commands
import org.bukkit.entity.Player


/**
 * Reloads all LevelledMobs configuration from disk
 *
 * @author lokka30
 * @since 2.0
 */
object ReloadSubcommand : CommandBase("levelledmobs.command.reload") {
    override val description = "Reloads LevelledMobs config files."

    fun buildCommand() : LiteralCommandNode<CommandSourceStack>{
        return createLiteralCommand("reload")
            .executes { ctx ->
                val main = LevelledMobs.instance
                val sender = ctx.source.sender
                main.reloadLM(sender)

                if (main.mainCompanion.hadRulesLoadError && sender is Player)
                    sender.sendMessage(FileLoader.getFileLoadErrorMessage())

                return@executes Command.SINGLE_SUCCESS
            }
            .build()
    }
}